zmccauley/wp-battlenet-plugin
# local environment setup

## Requirements
- Install the [Dev Containers](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers) exentsion in VS Code
- [Docker](https://www.docker.com/)
- [TablePlus](https://tableplus.com/)
    - This is a freemium DB product. For our use case, the free tier is good enough. You can just only have 2 open tabs at a time 

## Testing

- Pull down this repo somewhere in WSL

```bash
git clone git@github.com:zmccauley/wp-battlenet-plugin.git
```

- cd into the repo directory that was created

```bash
cd wp-battlenet-plugin
```

- Check out this branch

```bash
git checkout local-environment
```

- Open the repo in vscode
- If the dev containers extension is installed you will be prompted with the message below, click `Reopen in Container`
    - This is going to create a docker container that spins up wordpress and mounts the plugin into the `plugins` directory so it can be used within the test wordpress instance  

![Screenshot 2024-06-08 143055](https://github.com/zmccauley/wp-battlenet-plugin/assets/12040266/dcbe2ebb-9c7d-4654-ad24-e87ff407bd29)

- It may take a few minutes for the container to spin up, but once it does VS Code will relaunch from within the container. You'll get a directory structure like below and the highlighted section is where you'd make all your code edits to work on the plugin

![Screenshot 2024-06-08 145344](https://github.com/zmccauley/wp-battlenet-plugin/assets/12040266/2a68275d-0f68-4d5b-8982-09148ad03f4b)

- Now you should be able to open a browser and go to [http://localhost:8080](http://localhost:8080). You'll be met with the wordpress setup page. Fill out the forms to create a user and log into the admin afterwards. 
- Navigate to the [plugin admin page](http://localhost:8080/wp-admin/plugins.php) once you're logged in and active the `WP Battle.net Plugin` plugin.

![Screenshot 2024-06-08 145857](https://github.com/zmccauley/wp-battlenet-plugin/assets/12040266/0c9e724b-2ce3-4a5d-a22a-1c13b737a41e)

- 🎉 The plugin doesn't actually do anything yet, but you have a working local environment

- Let's set up table plus to connect to the database. Open tableplus and click the `+` to create a new connection and click `MySQL`

![Screenshot 2024-06-08 150351](https://github.com/zmccauley/wp-battlenet-plugin/assets/12040266/9bbf3335-55ea-4c02-86c0-0e41541e8a3d)

- Use the screen shot below as a reference to fill out the connection info. This info is created in the `docker-compose.yml`.
    - The password field is also `wordpress`

![Screenshot 2024-06-08 150719](https://github.com/zmccauley/wp-battlenet-plugin/assets/12040266/b619b847-aaf0-46c1-b5ba-9eead551b0ab)

- Hit the `test` button, it should say the connection is okay. Press okay
- 
![Screenshot 2024-06-08 150836](https://github.com/zmccauley/wp-battlenet-plugin/assets/12040266/4ebfbb1c-8960-424d-bc3d-45f046fc4ebc)

- Hit connect and you're in! You can click around to see some data in tables the wordpress generated during the setup screen

![Screenshot 2024-06-08 150930](https://github.com/zmccauley/wp-battlenet-plugin/assets/12040266/54e5a4d4-9296-4796-9839-9471ab26be8b)
